wcfsetup/install/files/lib/data/user/online/UserOnline.class.php

```diff
@@ -134,172 +134,7 @@ public function getBrowser() {
-		}
-		
-		// tapatalk 4
-		if (preg_match('~tapatalk/([\d\.]+)?~i', $this->userAgent, $match)) {
-			return 'Tapatalk '.(isset($match[1]) ? $match[1] : 4);
-		}
-		
-		// firefox
-		if (preg_match('~firefox/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Firefox '.$match[1];
-		}
-		
-		// maxthon
-		if (preg_match('~maxthon[ /]([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Maxthon '.$match[1];
-		}
-		
-		// iemobile
-		if (preg_match('~iemobile[ /]([\d\.]+)|MSIE ([\d\.]+).*XBLWP7~i', $this->userAgent, $match)) {
-			return 'Internet Explorer Mobile '.(isset($match[2]) ? $match[2] : $match[1]);
-		}
-		
-		// ie
-		if (preg_match('~msie ([\d\.]+)|Trident\/\d{1,2}.\d{1,2}; .*rv:([0-9]*)~i', $this->userAgent, $match)) {
-			return 'Internet Explorer '.(isset($match[2]) ? $match[2] : $match[1]);
-		}
-		
-		// edge
-		if (preg_match('~edge?/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Microsoft Edge '.$match[1];
-		}
-		
-		// edge mobile
-		if (preg_match('~edga/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Microsoft Edge Mobile '.$match[1];
-		}
-		
-		// vivaldi
-		if (preg_match('~vivaldi/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Vivaldi '.$match[1];
-		}
-		
-		// iron
-		if (preg_match('~iron/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Iron '.$match[1];
-		}
-		
-		// coowon
-		if (preg_match('~coowon/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Coowon '.$match[1];
-		}
-		
-		// coolnovo
-		if (preg_match('~(?:coolnovo|chromeplus)/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'CoolNovo '.$match[1];
-		}
-		
-		// yandex
-		if (preg_match('~yabrowser/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Yandex '.$match[1];
-		}
-		
-		// midori
-		if (preg_match('~midori/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Midori '.$match[1];
-		}
-		
-		// chrome mobile
-		if (preg_match('~(?:crios|crmo)/([\d\.]+)|chrome/([\d\.]+).*mobile~i', $this->userAgent, $match)) {
-			return 'Chrome Mobile '.(isset($match[2]) ? $match[2] : $match[1]);
-		}
-		
-		// kindle
-		if (preg_match('~kindle/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Kindle '.$match[1];
-		}
-		
-		// silk
-		if (preg_match('~silk/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Silk '.$match[1];
-		}
-		
-		// android browser
-		if (preg_match('~Android ([\d\.]+).*AppleWebKit~i', $this->userAgent, $match)) {
-			return 'Android Browser '.$match[1];
-		}
-		
-		// safari mobile
-		if (preg_match('~([\d\.]+) Mobile/\w+ safari~i', $this->userAgent, $match)) {
-			return 'Safari Mobile '.$match[1];
-		}
-		
-		// chrome
-		if (preg_match('~(?:chromium|chrome)/([\d\.]+)~i', $this->userAgent, $match)) {
-			return 'Chrome '.$match[1];
-		}
-		
-		// safari
-		if (preg_match('~([\d\.]+) safari~i', $this->userAgent, $match)) {
-			return 'Safari '.$match[1];
-		}
-		
-		return $this->userAgent;
+		return UserUtil::getBrowser($this->userAgent);
```

Comment:
Should this method be marked deprecated now?

Reply:
No, because it works similar to the method getFormattedIPAddress() in all classes. I only removed and outsourced the logic there, so that I can use it elsewhere.

Reply:
You are entirely correct. I somehow totally misread that code.